### PR TITLE
[NFC][Clang][RISCV] Fix typos of riscv-v-spec doc in  riscv_vector.td

### PR DESCRIPTION
**Pull request by punkyc**
Fix index typos, s.t. indexes in comments be same with riscv-v-spec doc.

### CODE REVIEW

File: clang/include/clang/Basic/riscv_vector.td [hunk 1]
```diff
@@ -1660,29 +1660,29 @@ let HasMasked = false, MaskedPolicyScheme = NonePolicy in {
...
+// 11.5. Vector Bitwise Logical Instructions
 let UnMaskedPolicyScheme = HasPassthruOperand in {
 defm vand : RVVIntBinBuiltinSet;
 defm vxor : RVVIntBinBuiltinSet;
 defm vor : RVVIntBinBuiltinSet;
 }
 defm vnot_v : RVVPseudoVNotBuiltin<"vxor", "csil">;
 
-// 12.6. Vector Single-Width Bit Shift Instructions
+// 11.6. Vector Single-Width Bit Shift Instructions
```
Comment (dtcxzyw) [inline clang/include/clang/Basic/riscv_vector.td:1671]: 11.6. Vector Single-Width Shift Instructions

File: clang/include/clang/Basic/riscv_vector.td [hunk 2]
```diff
@@ -1660,29 +1660,29 @@ let HasMasked = false, MaskedPolicyScheme = NonePolicy in {
...
+// 11.7. Vector Narrowing Integer Right Shift Instructions
 defm vnsrl : RVVUnsignedNShiftBuiltinSet;
 defm vnsra : RVVSignedNShiftBuiltinSet;
 }
 defm vncvt_x_x_w : RVVPseudoVNCVTBuiltin<"vnsrl", "vncvt_x", "csi",
                                          [["v", "vw"],
                                           ["Uv", "UvUw"]]>;
 
-// 12.8. Vector Integer Comparison Instructions
+// 11.8. Vector Integer Comparison Instructions
```
Comment (dtcxzyw) [inline clang/include/clang/Basic/riscv_vector.td:1685]: 11.8. Vector Integer Compare Instructions

File: clang/include/clang/Basic/riscv_vector.td [hunk 3]
```diff
@@ -2262,11 +2262,11 @@ defm vmfgt : RVVFloatingMaskOutBuiltinSet;
 defm vmfge : RVVFloatingMaskOutBuiltinSet;
 }
 
-// 14.14. Vector Floating-Point Classify Instruction
+// 13.14. Vector Floating-Point Classify Instruction
 let Name = "vfclass_v", UnMaskedPolicyScheme = HasPassthruOperand in
   def vfclass : RVVOp0Builtin<"Uv", "Uvv", "xfd">;
 
-// 14.15. Vector Floating-Point Merge Instructio
+// 13.15. Vector Floating-Point Merge Instructio
```
Comment (dtcxzyw) [inline clang/include/clang/Basic/riscv_vector.td:2269]: 13.15. Vector Floating-Point Merge Instruction

File: clang/include/clang/Basic/riscv_vector.td (hunk 1 above)
Comment (punkyc) [reply clang/include/clang/Basic/riscv_vector.td]: Fix it.

File: clang/include/clang/Basic/riscv_vector.td (hunk 2 above)
Comment (punkyc) [reply clang/include/clang/Basic/riscv_vector.td]: Fix it.

File: clang/include/clang/Basic/riscv_vector.td (hunk 3 above)
Comment (punkyc) [reply clang/include/clang/Basic/riscv_vector.td]: Fix it.

**Review by dtcxzyw — APPROVED** [review]
LGTM

**Pull request merged**